src/main/java/com/iota/iri/network/FIFOCache.java

```diff
@@ -0,0 +1,40 @@
+package com.iota.iri.network;
+
+import java.security.SecureRandom;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+
+public class FIFOCache<K, V> {
```

Comment:
would be great to move away from this non-generic impl. and replace it with a known Impl.
(it originally was an LRU cache with special requirements, but it's not the case anymore.)

Reply:
Maybe we could use a Cuckoo filter?

Reply:
Opened https://github.com/iotaledger/iri/issues/1456

I was going to write no need to do it now, but I see you partially did it :-P